### PR TITLE
CORTX-28862: added server_instances key for s3 in config 

### PR DESCRIPTION
**Pull request by kailashyadav11**
# Problem Statement
- Verify the conf csample files are proper and can be used for deployment

# Design
-  https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/651132980/CORTX+GConf+Config+Store+Keys
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: conf/gconf/cluster.conf.sample [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+      - https://motr-hax-svc:22003              # Control endpoint
+      - tcp://data1-node1:22001                 # For motr and Hax communication
+      - tcp://data1-node2:22001                 # For motr and Hax communication
+      - tcp://data1-node3:22001                 # For motr and Hax communication
+      - tcp://server-node1:22001                # if s3 client is configured
+      - tcp://server-node2:22001                # if s3 client is configured
+      - tcp://server-node3:22001                # if s3 client is configured
+      - tcp://client-node1:22001               # Only if motr_client is configured
+      - tcp://client-node2:22001               # Only if motr_client is configured
+      - tcp://client-node3:22001               # Only if motr_client is configured
```
Comment (tanujashinde0405) [inline conf/gconf/cluster.conf.sample:146]: 3 new endpoints has been added in page : https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/611615649/CORTX+Configuration+Parameters  ,do add them here also.

File: conf/solution/config.yaml.sample [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+          min: 300Mi
+          max: 2Gi
+        cpu:
+          min: 250m
+          max: 1000m
+  motr:
+    client_instances: 0                        # deprecated [part of client]
+    interface_type: tcp                        # deprecated [part of URL]
+    interface_family: inet                     # Optional: inet (default) | inet6
+    transport_type: libfab                     # libfab | lnet (libfab==inet)
```
Comment (tanujashinde0405) [inline conf/solution/config.yaml.sample:73]: md_size key is missing here

File: conf/solution/config.yaml.sample [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      - https://motr-hax-svc:22003              # Control endpoint
+      - tcp://data1-node1:22001                 # For motr and Hax communication
+      - tcp://data1-node2:22001                 # For motr and Hax communication
+      - tcp://data1-node3:22001                 # For motr and Hax communication
+      - tcp://server-node1:22001                # if s3 client is configured
+      - tcp://server-node2:22001                # if s3 client is configured
+      - tcp://server-node3:22001                # if s3 client is configured
+      - tcp://client-node1:22001                # Only if motr_client is configured
+      - tcp://client-node2:22001                # Only if motr_client is configured
+      - tcp://client-node3:22001                # Only if motr_client is configured
```
Comment (tanujashinde0405) [inline conf/solution/config.yaml.sample:128]: add newly added 3 data endpoints here also.

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]
please resolve merge conflicts.

File: conf/gconf/cluster.conf.sample [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+    security:
+      device_certificate: /etc/cortx/solution/ssl/stx.pem
+      domain_certificate: /etc/cortx/solution/ssl/stx.pem
+      ssl_certificate: /opt/seagate/cortx/s3/install/haproxy/ssl/s3.seagate.com.pem
+    service:
+      admin: admin
+      secret: !!binary |
+        Z0FBQUFBQmhVY1dCdzJ6RkRXT3FSVjJXTWtUM0paZlB1LXFRRW0xNy1rRGhJLVpHUjcxMkVOeDZ6
+        MG9KWmpsM21yTmprZTJkYTBHUU1OWVl4dnZSaVRTeEs4UEdReHhkR1E9PQ==
+    setup_size: small                               # To be removed and derived from limits
```
Comment (nitin-seagate) [inline conf/gconf/cluster.conf.sample:45]: Lets be consistent, mention all old keys as `deprecated`

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

File: conf/gconf/cluster.conf.sample (hunk 1 above)
Comment (kailashyadav11) [reply conf/gconf/cluster.conf.sample:146]: done

File: conf/solution/config.yaml.sample (hunk 2 above)
Comment (kailashyadav11) [reply conf/solution/config.yaml.sample:73]: done

File: conf/solution/config.yaml.sample (hunk 3 above)
Comment (kailashyadav11) [reply conf/solution/config.yaml.sample:129]: done

File: conf/gconf/cluster.conf.sample (hunk 4 above)
Comment (kailashyadav11) [reply conf/gconf/cluster.conf.sample]: done

**Review by tanujashinde0405 — APPROVED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Pull request merged**